assets/js/src/core/modules/notifications/hooks/use-notifications.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * This source file is available under the terms of the
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+import { useCallback, useMemo, useState } from 'react'
+import { type NotificationGetCollectionApiArg, type NotificationListItem, useNotificationGetCollectionMutation } from '../notifications-slice.gen'
+import { isNil } from 'lodash'
+import trackError, { ApiError, GeneralError } from '@Pimcore/modules/app/error-handler'
+
+interface UseNotificationsReturn {
+  getAllNotifications: () => Promise<void>
+  totalItems: number
+  notifications: NotificationListItem[]
+  isLoading: boolean
+  page: number
+  setPage: (page: number) => void
+  pageSize: number
+  setPageSize: (pageSize: number) => void
+}
+
+export const useNotifications = (): UseNotificationsReturn => {
+  const [page, setPage] = useState<number>(1)
+  const [pageSize, setPageSize] = useState(20)
+  const [notifications, setNotifications] = useState<NotificationListItem[]>([])
+  const [totalItems, setTotalItems] = useState<number>(0)
+
+  const queryArgs: NotificationGetCollectionApiArg = useMemo(() => ({ body: { filters: { page, pageSize, includeDescendants: true } } }), [page, pageSize])
+
+  const [requestNotifications, { isLoading }] = useNotificationGetCollectionMutation()
+
+  const getAllNotifications = useCallback(async (): Promise<void> => {
```

Comment:
This `useNotifications` hook feels a bit like it is not optimal. It's a bit strange that `getAllNotifications` is a method which then sets a state and the state variables are returned directly from the hook. So when the `getAllNotifications` hook does not get called the other state variables are useless. Either this hook should be responsible for getting all notifications only (and then directly fetch and return them without additional method call) or `getAllNotifications` should work somehow isolated. I'd suggest to just use the hook for the listing and remove the `getAllNotifications` method.